libres/lib/analysis/ies/ies.cpp

```diff
@@ -80,35 +80,33 @@ void ies::init_update(ies::data::Data &module_data,
     module_data.update_obs_mask(obs_mask);
 }
 
-void ies_initX__(const matrix_type *A, const matrix_type *Y0,
-                 const matrix_type *R, const matrix_type *E,
-                 const matrix_type *D, matrix_type *X,
+void ies_initX__(const std::optional<Eigen::MatrixXd> &A, const Eigen::MatrixXd &Y0,
```

Comment:
You mentioned the other day that this and other functions should return a matrix instead of us passing a reference to a matrix.

If I understood @pinkwah correctly, there is some guarantee that no copying will take place.
This seems to be a decent explanation:

https://stackoverflow.com/a/15528871/10731388